### PR TITLE
e2e Tests: Add more user auto-completer (mentions) coverage

### LINKED ISSUE #30632: e2e Tests: Cover user auto-completer
We should add an e2e test to cover the _user_ autocompleter (triggered by entering `@`). AFAICS, we don't have any such tests yet, so we'll likely need to add a new file.

Specifically, we should make sure that the following scenarios (raised by @gziolo in https://github.com/WordPress/gutenberg/pull/30540#issuecomment-814636881) work:

> As a follow-up, It would be great to add [...] test scenarios that cover:
> - `@Noah Allen` in the middle of the sentence
> - `@noahtallen @WunderBart` at any position

In other words: We need the e2e test to create a test user with a first and last name (e.g. `John Doe`), and need to make sure that typing `@John Doe` resolves to their username (e.g. `@johndoe`).

Furthermore, we need to make sure that entering _two_ usernames (each time starting with `@`) works.

cc/ @gziolo @fullofcaffeine @WunderBart

### PR DESCRIPTION
**Pull request by ockham**
## Description
Fixed #30632. See there for more background.

This adds two new helper functions to `e2e-test-utils`: `createUser` and `deleteUser`.

## How has this been tested?
To run locally:

```
npm run test-e2e -- packages/e2e-tests/specs/editor/various/mentions.test.js
```

To see what the test is doing, append ` --puppeteer-interactive` to that command.

### CODE REVIEW

File: packages/e2e-tests/specs/editor/various/mentions.test.js [hunk 1]
```diff
@@ -20,4 +31,27 @@ describe( 'autocomplete mentions', () => {
...
+
+	it( 'should insert mention between two other words', async () => {
+		await clickBlockAppender();
+		await page.keyboard.type( 'Stuck in the middle with you.' );
+		await pressKeyTimes( 'ArrowLeft', 'you.'.length );
+		await page.keyboard.type( '@j' );
+		await page.waitForSelector( '.components-autocomplete__result' );
+		await page.keyboard.press( 'Enter' );
+		await page.keyboard.type( ' ' );
+		expect( await getEditedPostContent() ).toMatchSnapshot();
```
Comment (gziolo) [inline packages/e2e-tests/specs/editor/various/mentions.test.js:43]: We could experiment here and in the next test with [`toMatchInlineSnapshot`](https://jestjs.io/docs/expect#tomatchinlinesnapshotpropertymatchers-inlinesnapshot) to make tests more readable.

**Review by gziolo — APPROVED** [review]
Nice one. This is a great addition for test utils, and a nice tests to have for that user auto-complete 💯

File: packages/e2e-tests/specs/editor/various/mentions.test.js (hunk 1 above)
Comment (gziolo) [reply packages/e2e-tests/specs/editor/various/mentions.test.js:43]: @ntsekouras, have you tried using inline snapshots?

Comment (ntsekouras) [reply packages/e2e-tests/specs/editor/various/mentions.test.js:43]: Oh... I'm sorry Grzegorz but for some reason I thought it was suggestion to have in mind for `next tests` :). I read about it and just thought that it was cool 😄 .

I created a small PR to update per your suggestion here: https://github.com/WordPress/gutenberg/pull/32727

**Pull request merged**